src/popup/components/form/FormAccountInput.vue

```diff
@@ -0,0 +1,246 @@
+    textLimit: { type: Number, default: null },
+    resizable: { type: Boolean, default: true },
+    size: {
+      type: String,
+      default: null,
+      validator: (val: string) => SIZES.includes(val),
+    },
+    options: {
+      type: Array,
+      default() {
+        return [];
+      },
+    },
+    enterSubmit: Boolean,
+    autoHeight: Boolean,
+    readonly: Boolean,
+    hideClearIcon: { type: Boolean, default: false },
+    protocol: { type: String as PropType<Protocol>, default: null },
+  },
+  emits: ['update:modelValue', 'submit'],
+  setup(props, { emit }) {
+    // const addressBook = useStorageRef<IAddressBookEntry[]>([], STORAGE_KEYS.addressBook);
+    const {
+      addressBookFiltered,
+      protocolFilter,
+    } = useAddressBook();
+    protocolFilter.value = props.protocol;
+
+    const selectedAddress = ref<IAddressBookEntry | undefined>();
+
+    const textarea = ref<HTMLTextAreaElement>();
+    const height = ref<string | undefined>();
+    function handleInput(event: InputEvent) {
+      const { value } = event.target as HTMLInputElement;
+      selectedAddress.value = undefined;
+      const searchAddress = addressBookFiltered.value
+        ?.find((entry) => (entry.address === value));
+      if (searchAddress) {
+        selectedAddress.value = searchAddress;
+        emit('update:modelValue', searchAddress.address);
+      } else {
+        emit('update:modelValue', value);
+      }
+    }
+
+    function handleUpdate(payload: IAddressBookEntry) {
+      selectedAddress.value = payload;
+      emit('update:modelValue', payload.address);
+    }
+
+    function handleEnterClick() {
+      if (props.enterSubmit) {
+        emit('submit');
+      }
+    }
+
+    function clearAddress() {
+      selectedAddress.value = undefined;
+      emit('update:modelValue', '');
+    }
+
+    onMounted(() => {
+      watch(() => props.modelValue, () => {
+        const searchAddress = addressBookFiltered.value
+          ?.find((entry: IAddressBookEntry) => (entry.address === props.modelValue));
+        if (searchAddress) {
+          selectedAddress.value = searchAddress;
+        }
+        if (!props.modelValue) { // If it's cleared externally
+          selectedAddress.value = undefined;
+        }
+
+        if (props.autoHeight && textarea.value) {
+          height.value = 'auto';
+          nextTick(() => {
+            if (!textarea.value) return;
+            const { scrollHeight, clientHeight } = textarea.value!;
+            const newHeight = clientHeight > scrollHeight ? clientHeight : scrollHeight;
+            height.value = `${newHeight}px`;
+          });
+        }
+      }, { immediate: true });
+    });
+
+    return {
+      textarea,
+      height,
+      addressBookFiltered,
+      handleInput,
+      handleUpdate,
+      handleEnterClick,
+      selectedAddress,
+      clearAddress,
+      CircleCloseIcon,
+    };
+  },
+  computed: {
+    PROTOCOLS() {
+      return PROTOCOLS;
+    },
+  },
```

Comment:
Please avoid using options API. Instead use the `computed` function imported from `vue` lib.